Fix: SQL compatibility, increase size for incoterms/code

mdeweerd opened the pull request:
# Fix: SQL compatibility, increase size for incoterms/code
    
- Modify llx_c_incoterms table to increase code field length
- Use `INSERT IGNORE` or `ON CONFLICT` depending on db type

mdeweerd committed:
Fix: SQL compatibility, increase size for incoterms/code

- Modify llx_c_incoterms table to increase code field length
- Use `INSERT IGNORE` or `ON CONFLICT` depending on db type

hregis commented:
@mdeweerd INSERT IGNORE is not compatible with Postrgresql

eldy commented on htdocs/install/mysql/migration/22.0.0-23.0.0.sql:
```diff
@@ -271,30 +271,41 @@ ALTER TABLE llx_blockedlog ADD COLUMN linktype varchar(16);
 ALTER TABLE llx_blockedlog ADD COLUMN vat double(24,8) DEFAULT NULL;
 
 
+WHERE code = 'CIP';
```
sql error here.

mdeweerd replied:
Fixed in update.

eldy commented on htdocs/install/mysql/migration/22.0.0-23.0.0.sql:
```diff
@@ -271,30 +271,41 @@ ALTER TABLE llx_blockedlog ADD COLUMN linktype varchar(16);
+ALTER TABLE llx_c_incoterms MODIFY code varchar(8) NOT NULL;
+
+-- For MySQL and MariaDB:
+-- VMYSQL5.7 INSERT IGNORE INTO llx_c_incoterms (code, label, libelle, active) VALUES ('DPU', 'Delivered at Place Unloaded', 'Delivered at Place Unloaded, marchandises déchargées et livrées au lieu de destination désigné (remplace DAT, élargit les lieux de livraison possibles)', 1);
```
You should be able to do a simple 
INSERT INTO llx_c_incoterms (code, label, libelle, active) VALUES ('DTP', 'Delivered at Terminal Paid', 'Delivered at Terminal Paid, marchandises livrées et dédouanées dans un terminal du pays de destination', 0);
If record already exist, it wil be trapped and ignore by the script upgrade.php that run the .sql

mdeweerd committed:
Simple sql INSERTs instead of INSERT IGNORE

mdeweerd committed:
Qual: Update Incoterms code field in table definition

Updated the llx_c_incoterms table to modify the code field length, and removed modification in the data file.

mdeweerd commented:
@hregis  Indeed - so I also removed the `ALTER TABLE` from `htdocs/install/mysql/data/llx_c_incoterms.sql`.
